moveit_ros/planning_interface/moveit_cpp/include/moveit/moveit_cpp/planning_component.h

```diff
@@ -0,0 +1,216 @@
+
+  /** \brief Destructor */
+  ~PlanningComponent();
+
+  /** \brief Get the name of the planning group */
+  const std::string& getName() const;
+
+  /** \brief Get the names of the named robot states available as targets */
+  const std::vector<std::string> getNamedTargets();
+
+  /** \brief Get the joint values for targets specified by name */
+  std::map<std::string, double> getNamedTargetValues(const std::string& name);
+
+  /** \brief Specify the workspace bounding box.
+       The box is specified in the planning frame (i.e. relative to the robot root link start position).
+       This is useful when the planning group contains the root joint of the robot -- i.e. when planning motion for the
+     robot relative to the world. */
+  void setWorkspace(double minx, double miny, double minz, double maxx, double maxy, double maxz);
+
+  /** \brief Remove the workspace bounding box from planning */
+  void unsetWorkspace();
+
+  /** \brief Get the considered start state if defined, otherwise get the current state */
+  robot_state::RobotStatePtr getStartState();
+
+  /** \brief Set the robot state that should be considered as start state for planning */
+  bool setStartState(const robot_state::RobotState& start_state);
+  /** \brief Set the named robot state that should be considered as start state for planning */
+  bool setStartState(const std::string& named_state);
+
+  /** \brief Set the start state for planning to be the current state of the robot */
+  void setStartStateToCurrentState();
+
+  /** \brief Set the goal constraints used for planning */
+  bool setGoal(const std::vector<moveit_msgs::Constraints>& goal_constraints);
+  /** \brief Set the goal constraints generated from a target state */
+  bool setGoal(const robot_state::RobotState& goal_state);
+  /** \brief Set the goal constraints generated from target pose and robot link */
+  bool setGoal(const geometry_msgs::PoseStamped& goal_pose, const std::string& link_name);
+  /** \brief Set the goal constraints generated from a named target state */
+  bool setGoal(const std::string& named_target);
+
+  /** \brief Run a plan from start or current state to fulfill the last goal constraints provided by setGoal() using
+   * default parameters. */
+  PlanSolution plan();
+  /** \brief Run a plan from start or current state to fulfill the last goal constraints provided by setGoal() using the
+   * provided PlanRequestParameters. */
+  PlanSolution plan(const PlanRequestParameters& parameters);
+
+  /** \brief Execute the latest computed solution trajectory computed by plan(). By default this function terminates
+   * after the execution is complete. The execution can be run in background by setting blocking to false. */
+  bool execute(bool blocking = true);
+
+  /** \brief Return the last plan solution*/
+  const PlanSolutionPtr getLastPlanSolution();
+
+private:
+  // Core properties and instances
+  ros::NodeHandle nh_;
+  MoveItCppPtr moveit_cpp_;
+  const std::string group_name_;
```

Comment:
do we need to store this string, or can we just use joint_model_group_->getName()?